### PR TITLE
Remove type literals and rationalize type.js

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: runtime/particle-spec.js [hunk 1]
```diff
@@ -38,7 +38,13 @@ class ConnectionSpec {
 class ParticleSpec {
   constructor(model, resolveSchema) {
     // TODO: This should really happen after parsing, not here.
-    model.args.forEach(arg => arg.type = Type.resolveSchemas(arg.type, resolveSchema));
+    assert(model.args);
+    for(var arg of model.args) {
+      if (arg.type.resolveSchemas == undefined)
+        console.log(arg.type);
```
Comment (dstockwell) [inline runtime/particle-spec.js]: remove

File: runtime/particle-spec.js [hunk 2]
```diff
@@ -38,7 +38,13 @@ class ConnectionSpec {
...
+      arg.type = arg.type.resolveSchemas(resolveSchema);
+    }
+    // model.args.forEach(arg => {console.log(arg.type.resolveSchemas); arg.type = arg.type.resolveSchemas(resolveSchema)});
```
Comment (dstockwell) [inline runtime/particle-spec.js]: remove

File: runtime/test/arc-tests.js [hunk 3]
```diff
@@ -30,6 +30,7 @@ describe('Arc', function() {
 
   it('applies existing runtime to a particle', async () => {
     let arc = new Arc({loader, slotComposer});
+    console.log(Foo.type, Foo.type.isEntity);
```
Comment (dstockwell) [inline runtime/test/arc-tests.js]: remove

File: runtime/type.js [hunk 4]
```diff
@@ -8,136 +8,144 @@
...
+    return !this.variableReference;
   }
 
   primitiveType() {
-    return new Type(typeLiteral.primitiveType(this.key));
-  }
-  get schema() {
-    assert(this.isEntity);
-    return this.key.schema;
+    var type = this.viewType;
```
Comment (dstockwell) [inline runtime/type.js]: should this assert that it's a view?

File: runtime/particle-spec.js (hunk 1 above)
Comment (shans) [inline runtime/particle-spec.js]: Done

File: runtime/particle-spec.js (hunk 2 above)
Comment (shans) [inline runtime/particle-spec.js]: Switched to this instead.

File: runtime/test/arc-tests.js (hunk 3 above)
Comment (shans) [inline runtime/test/arc-tests.js]: Done

File: runtime/type.js (hunk 4 above)
Comment (shans) [inline runtime/type.js]: There's an assert in the viewType accessor.

**Pull request merged**